providers/utils.ts

```diff
@@ -14,12 +14,16 @@ import { OAuthClient } from '../services/client'
 export const Logger = ServiceProvider.createSymbol<GenericLogger>('Logger')
 export const Serializer =
   ServiceProvider.createSymbol<GenericSerializer>('Serializer')
+
 export const HttpClient =
   ServiceProvider.createSymbol<GenericHttpClient>('HttpClient')
 
+export const MemoryCache =
+  ServiceProvider.createSymbol<GenericMemoryCache>('MemoryCache')
+
 const utilsProvider = new ServiceProvider()
 
-utilsProvider.register(HttpClient, Lifecycle.Singleton, () => new AxiosClient())
+utilsProvider.register(HttpClient, Lifecycle.Transient, () => new AxiosClient())
```

Comment:
```suggestion
utilsProvider.register(HttpClient, Lifecycle.Singleton, () => new AxiosClient())
```

Maybe you could change this back into a singleton? 🙂

Reply:
I don't think it can be a singleton as interceptors are assigned to an instance to keep track of the authorization headers. Which are different per consumer

Reply:
Makes sense, maybe we could make the interceptors part of the HttpClient interface? I'd like to break the dependency on the Axios package. Would that be feasible for this PR? 🙂